### ISSUE TITLE
Is it recommended to persist the raftDir?

### ISSUE THREAD

**Issue by munnerz** [Boostport/kubernetes-vault#60]
I've got this project set up and working great - I've just spotted however, that `raftDir` is mentioned in the README as the directory where raft data is stored.

Does the data in this directory need to be persisted anywhere? What's the effect of not persisting it? Could it potentially lead to issues during DR scenario's?

I'd be happy to make a PR on the docs to clarify this if you could fill me in!

Thanks 😄

**Comment by F21** [comment]
Hey,

Good to hear you've success with it 😄 

It only uses raft to elect a leader because only 1 instance should be pushing the wrapped `secret_id`s to pods at any time. Other than that, the `kubernetes-vault` controller does not store nor create any important data at all.

I can think of the following failure modes:
- A `kubernetes-vault` controller instance or a kubernetes node on which it is running on fails: Kubernetes will reschedule a new instance, it will discover the other existing instances using gossip and then replicate the current raft log from those instances.

- The whole Kubernetes cluster goes down: In this case, we do not care about the data in the raft log. When Kubernetes comes back up, it will restart those `kubernetes-vault` instances and they will see each other and start a new raft log and elect a new leader.

The only thing you should be aware with the second failure mode is that the Vault periodic token could expire during the outage. Possible solutions are:
- Set the periodic token's period to be more than the maximum outage time of your cluster.
- Manually intervene and set a new Vault periodic token to the Kubernetes deployment of the `kubernetes-vault` controller before bringing `kubernetes-vault` cluster back online.

tldr;
The raft log does not contain any important data and can be thrown away.

If you wish to make a PR to clarify this, I am happy to take a look! 😄

**Comment by F21** [comment]
@munnerz Can this be closed?

**Comment by munnerz** [comment]
Hey @F21 

Sorry for not responding sooner - thanks very much for the clarification! Extremely helpful 😄 

I will try and find some time put this into some documentation soon - currently quite a busy time though!!

Yep I'll close it now.
